### PR TITLE
Added 0 principle

### PR DESCRIPTION
**Pull request by solarkennedy**
This adds the whole "why do this in the first place" answer, which is kinda the 0'th principle of PaaSTA. This doc kinda assumes that you already want to build a paas yourself, so I thought we might make this 0th principle explicit in this doc?

### CODE REVIEW

File: docs/source/about/paasta_principles.rst [hunk 1]
```diff
@@ -14,6 +14,35 @@ for the exact contract an app must meet to run on PaaSTA is documented in the
...
+own, even if it makes heavy use of existing components (like PaaSTA), should
+not be taken lightly.
+
+On the plus side, building your own PaaS can give your organization more value
+than adopting an existing commercial or third-party PaaS. The value comes from
+the fact that you can build your PaaS to meet your specific needs like a glove.
+Additionally, the much of the existing engineering effort towards your current
+infrastructure can often be re-used. For example if you have an existing
+service-discovery mechanism, you can build a PaaS that integrates tightly
+with it, as opposed to replacing it with a difference service-discovery mechanism
```
Comment (hashbrowncipher) [inline docs/source/about/paasta_principles.rst]: different

File: docs/source/about/paasta_principles.rst [hunk 2]
```diff
@@ -14,6 +14,35 @@ for the exact contract an app must meet to run on PaaSTA is documented in the
...
+from a third-party.
+
+There are of course downsides to building your own PaaS. There is certainly
+some engineering effort in involved, but that must be weight against the
```
Comment (hashbrowncipher) [inline docs/source/about/paasta_principles.rst]: weighed

Comment (hashbrowncipher) [inline docs/source/about/paasta_principles.rst]: s/in //

File: docs/source/about/paasta_principles.rst [hunk 3]
```diff
@@ -14,6 +14,35 @@ for the exact contract an app must meet to run on PaaSTA is documented in the
...
+engineering effort involved in *not* building your own PaaS (integration,
+training, and migration costs). You also lose the support of the community,
+or in the case of a commercial PaaS, the support of a vendor. Problems with
+your own PaaS are not "googleable". No new employees will know how your PaaS
+works, you cannot hire existing experts on it.
+
+PaaSTA was built because we believed we could provide business value by
+building our own PaaS to seamlessly meld with Yelps existing infrastructure. At
```
Comment (hashbrowncipher) [inline docs/source/about/paasta_principles.rst]: Yelp's

File: docs/source/about/paasta_principles.rst (hunk 2 above)
Comment (hashbrowncipher) [inline docs/source/about/paasta_principles.rst]: I feel like 'some' here is a weasel word.  I would omit it in my writing; up to you.

File: docs/source/about/paasta_principles.rst [hunk 4]
```diff
@@ -14,6 +14,35 @@ for the exact contract an app must meet to run on PaaSTA is documented in the
...
+0. A Bespoke PaaS is not Necessarily a Bad Thing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Building your own PaaS is a non-trivial undertaking. The decision to build your
+own, even if it makes heavy use of existing components (like PaaSTA), should
+not be taken lightly.
+
+On the plus side, building your own PaaS can give your organization more value
+than adopting an existing commercial or third-party PaaS. The value comes from
+the fact that you can build your PaaS to meet your specific needs like a glove.
```
Comment (Rob-Johnson) [inline docs/source/about/paasta_principles.rst]: drop the 'like a glove'

File: docs/source/about/paasta_principles.rst [hunk 5]
```diff
@@ -14,6 +14,35 @@ for the exact contract an app must meet to run on PaaSTA is documented in the
...
+Additionally, the much of the existing engineering effort towards your current
```
Comment (Rob-Johnson) [inline docs/source/about/paasta_principles.rst]: s/the much/much/

**Pull request merged**